templates/content/node--oe-project--full.html.twig

```diff
@@ -4,46 +4,50 @@
  * Project full display.
  */
 #}
-{% if not content.oe_subject.isEmpty() %}
-  {% set _badges = [] %}
-  {% for _badge in content.oe_subject|field_value %}
-    {% set _badges = _badges|merge([{
-      label: _badge|raw,
-      rounded_pill: true,
-    }]) %}
-  {% endfor %}
-{% endif %}
-{{ pattern('content_banner', {
-  background: 'gray',
-  title: label,
-  content: content.oe_teaser,
-  image: image,
-  attributes: create_attribute().addClass(['ps-0']),
-  badges: _badges,
-}) }}
+{% apply spaceless %}
+
+{% macro header(title, id) %}
+  {% set attributes = id ? create_attribute({'id': id}) %}
+  <h3 class="fw-bold mb-4 mt-4-5"{{ attributes }}>{{ title }}</h3>
+{% endmacro %}
+
 {% macro field_with_header(field) %}
-  {% if field|field_value is not empty %}
-    <h3 class="fw-bold mb-4 pt-3" id="{{ field['#inpage_nav_id'] }}">{{ field|field_label }}</h3>
-    {{ field }}
+  {% set _content %}{{ field|field_value }}{% endset %}
+  {% if _content|trim %}
+    {{ _self.header(field|field_label, field['#inpage_nav_id'] ?? null) }}
+    {{ _content }}
   {% endif %}
 {% endmacro %}
+
 {% set _project_details %}
-  {{ content.group_project_details }}
+  {% if project_status_args is not empty %}
+    {{ attach_library('oe_whitelabel/project_status') }}
+    {% include '@oe-bcl/project-status' with project_status_args only %}
+  {% endif %}
+  {% if contributions_args is defined %}
+    {% include '@oe-bcl/project-status/project-contributions' with contributions_args only %}
+  {% endif %}
+  {{ content.group_project_details|filter((subgroup, key) => key|first != '#') }}
```

Comment:
This is element_children() for poor people.
We want to render each element in the group, but not the group wrapper. We want the wrapper around the entire project details including the status/progress and the contributions.

I would much prefer to have custom twig filters in oe_bootstrap_theme or elsewhere, see https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-1447.
Or a custom field group formatter "bare" that does not render any wrappers.

Alternatively I could render each item in the group separately.

Reply:
Not good to skip wrappers, because cache information can be there in some scenarios. Much better to use a bare template.

Reply:
I think for field groups all the relevant cache tags sit in the fields, so it would be safe like this. But I could be wrong :)

But it makes me wonder if the twig filters `|field_value` and `|field_label` from https://www.drupal.org/project/twig_field_value are safe for caching.
The module page has a cache warning about field_raw and field_target_entity but not about field_value and field_label.

I need to learn a bit more how cache tags are processed during the render process.

For the bare template: I don't like adding more and more specific templates. But something like this could work (this one is for a field, not a group):
```twig
content.body|merge({'#theme': 'field__bare'})
```

Reply:
I think we are safe here, as said. I will leave this one unchanged for now.

Reply:
Can't we just change the template suggestion in the node full preprocess?

Reply:
I did some debugging, and found that field group elements don't contain `'#cache'` entries.

Field group elements are populated in a preprocess on the node/entity itself, which simply moves fields around.

One thing we _could_ do is this:

```twig
{{ content.group_project_details|merge({'#theme_wrappers': []}) }}
```

This way the wrapper div would not be rendered, but the group element would still go through the render pipeline.
Perhaps in addition we also want to remove `'#attached'` data.
Obviously any processing applied by the field group formatter in the preprocess phase already happened before it hits the template.

A cleaner option would be to have a special field group formatter that does not apply any wrappers. See https://www.drupal.org/project/field_group/issues/3279137.
It seems straightforward, we already have one field group formatter.

I am not sure if I want to do this in the scope of this issue.